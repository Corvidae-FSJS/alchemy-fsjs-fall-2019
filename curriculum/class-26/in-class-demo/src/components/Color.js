import React from 'react';

const Color = () => (
  <dl>
    <dt>Name</dt>
    <dd>Blue <div style={{ width: '100px', height: '100px', backgroundColor: 'blue' }}></div></dd>

    <dt>Hex</dt>
    <dt>#0000FB</dt>

    <dt>RGB</dt>
    <dd>
      <p>Red - 0</p>
      <p>Green - 0</p>
      <p>Blue - 245</p>
    </dd>
  </dl>
);

export default Color;
