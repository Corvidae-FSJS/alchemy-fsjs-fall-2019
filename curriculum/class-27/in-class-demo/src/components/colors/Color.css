.Color {
  color: red;

  dt {
    color: blue;

    input {
      &:hover {
        color: yellow;
        transform: translateY(100px);
      }
    }
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
