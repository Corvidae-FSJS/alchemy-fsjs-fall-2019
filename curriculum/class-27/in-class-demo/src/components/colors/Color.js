import React from 'react';
import PropTypes from 'prop-types';
import styles from './Color.css';

const colorToHex = color => color.toString(16).padStart(2, '0').toUpperCase();

const rgbToHex = rgb => {
  return `#${colorToHex(rgb.r)}${colorToHex(rgb.g)}${colorToHex(rgb.b)}`;
};

const Color = ({ name, rgb }) => {
  const hex = rgbToHex(rgb);

  return (
    <dl className={styles.Color}>
      <dt><input type="text" /></dt>
      <dd>{name}</dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>RGD</dt>
      <dd>
        <p>Red - {rgb.r}</p>
        <p>Green - {rgb.g}</p>
        <p>Blue - {rgb.b}</p>
      </dd>

    </dl>
  );
};

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.shape({
    r: PropTypes.number.isRequired,
    g: PropTypes.number.isRequired,
    b: PropTypes.number.isRequired
  }).isRequired
};

export default Color;
