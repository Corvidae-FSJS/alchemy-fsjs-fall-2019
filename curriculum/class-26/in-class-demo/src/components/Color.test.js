import React from 'react';
import { shallow } from 'enzyme'
import Color from './Color';

describe('Color component', () => {
  it('matches a snapshot', () => {
    const color = shallow(<Color />);
    expect(color).toMatchSnapshot();
  });
});
