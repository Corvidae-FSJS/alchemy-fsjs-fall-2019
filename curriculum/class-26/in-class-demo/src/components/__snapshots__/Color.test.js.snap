// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component matches a snapshot 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Blue 
    <div
      style={
        Object {
          "backgroundColor": "blue",
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </dd>
  <dt>
    Hex
  </dt>
  <dt>
    #0000FB
  </dt>
  <dt>
    RGB
  </dt>
  <dd>
    <p>
      Red - 0
    </p>
    <p>
      Green - 0
    </p>
    <p>
      Blue - 245
    </p>
  </dd>
</dl>
`;
